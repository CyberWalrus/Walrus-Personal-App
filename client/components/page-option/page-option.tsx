import withDataState from "@client/hocs/with-data-state/with-data-state";
import { UserRole } from "@client/type/data";
import { FormType } from "@config/constants";
import { FunctionComponent, ReactElement } from "react";
import * as React from "react";
import { SortableContainer, SortableElement } from "react-sortable-hoc";
import { compose } from "recompose";
import Footer from "../footer/footer";
import FormCustom from "../form-custom/form-custom";
import Header from "../header/header";

export interface PropsHoc {
  userRoles: UserRole[];
  onSortEnd: () => void;
}
type Props = PropsHoc;

const SortableItem = SortableElement(
  ({name, id, isActive}: UserRole): ReactElement => (
    <div className={`data-container`}>
      <div>{name}</div>
      <div>{id}</div>
      <div>
        <input type={`checkbox`} defaultChecked={isActive} />
      </div>
    </div>
  ),
);
interface PropsSortableList {
  items: UserRole[];
}
const SortableList = SortableContainer(({items}: PropsSortableList) => {
  return (
    <fieldset className={`data-box`}>
      {items.map(({name, id, isActive}: UserRole, index: number) => (
        <SortableItem
          key={`item-${index.toString()}`}
          index={index}
          name={name}
          id={id}
          isActive={isActive}
        />
      ))}
    </fieldset>
  );
});

const PageOptions: FunctionComponent<Props> = ({
  userRoles,
  onSortEnd,
}: Props): ReactElement => {
  return (
    <div className={`page`}>
      <Header />
      <main className={`page-content`}>
        <section>
          <h2>Options</h2>
          <SortableList items={userRoles} onSortEnd={onSortEnd} />
        </section>
        <section className={`form-custom-box`}>
          <FormCustom formType={FormType.USER_ROLE} titel={`Add User Role`} />
        </section>
      </main>
      <Footer />
    </div>
  );
};

export { PageOptions };

const wrapper = compose(withDataState);

export default wrapper(PageOptions);
