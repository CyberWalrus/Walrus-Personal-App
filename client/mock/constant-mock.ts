import { FormType } from "../../config/constants";
import { Option } from "../hocs/with-form-state/with-form-state";
import { InputCustomInterface } from "../type/component";
import { User, UserRole } from "../type/data";

// tslint:disable-next-line:no-empty
const _functVoid = (): void => {};

const FORM_TYPE: FormType = FormType.SIGN_IN;

const user: User = {
  id: `test`,
  email: `test`,
  isActive: true,
  nickName: `test`,
  login: `test`,
  firstName: `test`,
  lastName: `test`,
  userRoleId: [`test`],
  signUpDate: 1234567,
  created: 1234567,
};
const userRole: UserRole = {
  id: `test`,
  name: `test`,
  isActive: true,
};
const optionForm: Option = {
  values: {},
  formErrors: {},
  formValid: true,
  onChangeUserInput: _functVoid,
  onClickSubmit: _functVoid,
};
const optionInput: InputCustomInterface = {
  formTypes: [FormType.SIGN_UP],
  type: `text`,
  name: `login`,
  placeHolder: `Login`,
  hidenValue: `Login`,
};

const usersEmpty: User[] = [];
const users: User[] = [user, user];

const userRolesEmpty: UserRole[] = [];
const userRoles: UserRole[] = [userRole, userRole];

export {
  user,
  usersEmpty,
  users,
  optionForm,
  FORM_TYPE,
  optionInput,
  userRole,
  userRolesEmpty,
  userRoles,
};
