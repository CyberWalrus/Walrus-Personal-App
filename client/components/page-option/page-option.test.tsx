import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson, { Json } from "enzyme-to-json";
import * as React from "react";
import { userRoles, userRolesEmpty } from "../../mock/constant-mock";
import { PageOptions } from "./page-option";

Enzyme.configure({ adapter: new Adapter() });
it(`PageOptions correctly render without data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(
      <PageOptions
        userRoles={userRolesEmpty}
        onSortEnd={jestFunc}
        onDelete={jestFunc}
      />,
    ),
  );
  expect(tree).toMatchSnapshot();
});
it(`PageOptions correctly render with data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(
      <PageOptions
        userRoles={userRoles}
        onSortEnd={jestFunc}
        onDelete={jestFunc}
      />,
    ),
  );
  expect(tree).toMatchSnapshot();
});
