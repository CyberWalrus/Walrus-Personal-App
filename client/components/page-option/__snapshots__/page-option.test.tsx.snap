// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageOptions correctly render with data 1`] = `
<div
  className="page"
>
  <Header />
  <main
    className="page-content"
  >
    <section>
      <h2>
        Options
      </h2>
      <sortableList
        axis="y"
        disableAutoscroll={false}
        distance={0}
        getHelperDimensions={[Function]}
        hideSortableGhost={true}
        items={
          Array [
            Object {
              "id": "test",
              "isActive": true,
              "name": "test",
            },
            Object {
              "id": "test",
              "isActive": true,
              "name": "test",
            },
          ]
        }
        lockOffset="50%"
        lockToContainerEdges={false}
        onDelete={[MockFunction]}
        onSortEnd={[MockFunction]}
        pressDelay={0}
        pressThreshold={5}
        shouldCancelStart={[Function]}
        transitionDuration={300}
        useWindowAsScrollContainer={false}
      />
    </section>
    <section
      className="form-custom-box"
    >
      <Connect(WithFormState)
        formType="USER_ROLE"
        titel="Add User Role"
      />
    </section>
  </main>
  <Footer />
</div>
`;

exports[`PageOptions correctly render without data 1`] = `
<div
  className="page"
>
  <Header />
  <main
    className="page-content"
  >
    <section>
      <h2>
        Options
      </h2>
      <sortableList
        axis="y"
        disableAutoscroll={false}
        distance={0}
        getHelperDimensions={[Function]}
        hideSortableGhost={true}
        items={Array []}
        lockOffset="50%"
        lockToContainerEdges={false}
        onDelete={[MockFunction]}
        onSortEnd={[MockFunction]}
        pressDelay={0}
        pressThreshold={5}
        shouldCancelStart={[Function]}
        transitionDuration={300}
        useWindowAsScrollContainer={false}
      />
    </section>
    <section
      className="form-custom-box"
    >
      <Connect(WithFormState)
        formType="USER_ROLE"
        titel="Add User Role"
      />
    </section>
  </main>
  <Footer />
</div>
`;
